### PR TITLE
ENH: Adds quantopian-quandl bundle as new default.

### PR DESCRIPTION
**Pull request by llllllllll**
This data bundle will use the quantopian mirror of the quandl WIKI data
instead of downloading from quandl directly. This dramatically improves
the speed because we do not pay the rate limiting for quandl and we can
send the data in the format zipline expects.

NOTE: I need to rebuild the quandl set in s3. To test locally you can change the path to point to `s3://.../test/` which is a smaller sample of data which works. I will run the load overnight tonight.

### CODE REVIEW

File: docs/source/beginner-tutorial.rst [hunk 1]
```diff
@@ -140,8 +140,8 @@ on OSX):
 
 As you can see there are a couple of flags that specify where to find your
 algorithm (``-f``) as well as parameters specifying which data to use,
-defaulting to the :ref:`quandl-data-bundle`. There are also arguments for the
-date range to run the algorithm over (``--start`` and ``--end``). Finally,
+defaulting to the :ref:`quantopian-quandl-mirror`. There are also arguments for
```
Comment (richafrank) [inline docs/source/beginner-tutorial.rst]: Is `mirror` part of the name somewhere?

Comment (llllllllll) [inline docs/source/beginner-tutorial.rst]: a ref link without an explicit link name will use the name of the next section. this renders as 'Quantopian Quandl WIKI Mirror': https://github.com/quantopian/zipline/pull/1178/files/265f77bf3e0ea53535fa6f025d44f5d05e0b2f48#diff-d2e46d2591b3fdb2539b3947867f60a5R106

Comment (richafrank) [inline docs/source/beginner-tutorial.rst]: Aha, awesome.

File: zipline/data/bundles/quandl.py [hunk 2]
```diff
@@ -291,8 +296,30 @@ def quandl_bundle(environ,
...
+                             daily_bar_writer,
+                             adjustment_writer,
+                             calendar,
+                             cache,
+                             show_progress,
+                             output_dir):
+    # use closing for py2 compat
+    with closing(urlopen(QUANTOPIAN_QUANDL_URL)) as f, \
+            tarfile.open('r', fileobj=BytesIO(f.read())) as tar:
+        tar.extractall(output_dir)
```
Comment (richafrank) [inline zipline/data/bundles/quandl.py]: Since this isn't instantaneous, would it be possible to show progress?

Comment (llllllllll) [inline zipline/data/bundles/quandl.py]: idk how to get a progress bar around this operation.

Comment (richafrank) [inline zipline/data/bundles/quandl.py]: Ok, too bad.

File: docs/source/bundles.rst [hunk 3]
```diff
@@ -16,10 +16,11 @@ could involve downloading and processing a lot of data. This can be run with:
 
 .. code-block:: bash
 
-   $ python -m zipline ingest <bundle>
+   $ python -m zipline ingest [-b <bundle>]
 
 
-where ``<bundle>`` is the name of the bundle to ingest.
+where ``<bundle>`` is the name of the bundle to ingest, defaulting to
+:ref:`quantipian-quandl <quantopian-quandl-mirror>`.
```
Comment (ssanderson) [inline docs/source/bundles.rst]: quantipian

**Comment by ssanderson** [comment]
The CLI crashed when I tried to use run from the command line with this:

```
$ python -m zipline ingest  # Wait 10 seconds or so.  Would be nice to have some progress indication here.
$ python -m zipline run -f zipline/examples/dual_moving_average.py -s 2014 -e 2015

Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/home/ssanderson/quantopian/qexec/zipline_repo/zipline/__main__.py", line 345, in <module>
    cli()
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/click/core.py", line 664, in __call__
    return self.main(*args, **kwargs)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/click/core.py", line 644, in main
    rv = self.invoke(ctx)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/click/core.py", line 991, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/click/core.py", line 837, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/click/core.py", line 464, in invoke
    return callback(*args, **kwargs)
  File "/home/ssanderson/quantopian/qexec/zipline_repo/zipline/__main__.py", line 95, in _
    return f(*args, **kwargs)
  File "/home/ssanderson/quantopian/qexec/zipline_repo/zipline/__main__.py", line 238, in run
    environ=os.environ,
  File "zipline/utils/run_algo.py", line 115, in _run
    bundle_timestamp,
  File "zipline/data/bundles/core.py", line 406, in load
    asset_db_path(name, timestr, environ=environ),
  File "zipline/assets/assets.py", line 108, in __init__
    metadata = sa.MetaData(bind=engine)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3381, in __init__
    self.bind = bind
  File "<string>", line 1, in <lambda>
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/sqlalchemy/sql/schema.py", line 3477, in _bind_to
    self._bind = sqlalchemy.create_engine(bind)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/sqlalchemy/engine/__init__.py", line 386, in create_engine
    return strategy.create(*args, **kwargs)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 49, in create
    u = url.make_url(name_or_url)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/sqlalchemy/engine/url.py", line 186, in make_url
    return _parse_rfc1738_args(name_or_url)
  File "/home/ssanderson/.virtualenvs/qexec/local/lib/python2.7/site-packages/sqlalchemy/engine/url.py", line 235, in _parse_rfc1738_args
    "Could not parse rfc1738 URL from string '%s'" % name)
sqlalchemy.exc.ArgumentError: Could not parse rfc1738 URL from string '/home/ssanderson/.zipline/data/quantopian-quandl/1462468523226428000/assets-3.sqlite'
```

**Comment by llllllllll** [comment]
@ssanderson ah, I can repro that in py2. weird that it would change, looking into that now

**Pull request merged**